ballerina/http_service_endpoint.bal

```diff
@@ -203,7 +203,8 @@ public type ListenerHttp1Settings record {|
 # Provides inbound request URI, total header and entity body size threshold configurations.
 #
 # + maxUriLength - Maximum allowed length for a URI. Exceeding this limit will result in a `414 - URI Too Long`
-#                  response
+#                  response. For HTTP/2, this limit will not be applicable as it already has a `:path`
```

Comment:
Isn't it better to make the existing one as a part of `ListenerHttp1Settings`?
```bal
public type Http1RequestLimitConfigs record {|
    int maxUriLength = 4096;
    int maxHeaderSize = 8192;
    int maxEntityBodySize = -1;
|};

public type ListenerHttp1Settings record {|
    KeepAlive keepAlive = KEEPALIVE_AUTO;
    int maxPipelinedRequests = MAX_PIPELINED_REQUESTS;
    Http1RequestLimitConfigs requestLimits;
|};
```

Then we can deprecate this `maxUriLength` field here.

Reply:
We can do that. However it will be confusing for the user when switching between http and http/2 right?
Defining `maxHeaderSize` for http/2 in default `http:ListenerConfiguration` and defining `maxHeaderSize` for http1 in `ListenerHttp1Settings`....

Reply:
Decided to keep the configs as it is because if we move the urilength request limit outside, the config `RequestLimitConfigs` will not be meaningful anymore.